airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/TimeStringToIntegral.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.data
+
+import io.airbyte.cdk.load.message.DestinationRecord
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.OffsetTime
+import java.time.ZoneOffset
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+
+class TimeStringTypeToIntegralType : AirbyteSchemaIdentityMapper {
+    override fun mapDate(schema: DateType): AirbyteType = DateTypeIntegral
+    override fun mapTimeTypeWithTimezone(schema: TimeTypeWithTimezone): AirbyteType =
+        TimeTypeIntegral
+    override fun mapTimeTypeWithoutTimezone(schema: TimeTypeWithoutTimezone): AirbyteType =
+        TimeTypeIntegral
+    override fun mapTimestampTypeWithTimezone(schema: TimestampTypeWithTimezone): AirbyteType =
+        TimestampTypeIntegral
+    override fun mapTimestampTypeWithoutTimezone(
+        schema: TimestampTypeWithoutTimezone
+    ): AirbyteType = TimestampTypeIntegral
+}
+
+/**
+ * NOTE: To keep parity with the old avro/parquet code, we will always first try to parse the value
+ * as with timezone, then fall back to without. But in theory we should be more strict.
+ */
+class TimeStringToIntegral(meta: DestinationRecord.Meta) : AirbyteValueIdentityMapper(meta) {
+    companion object {
+        private val DATE_TIME_FORMATTER: DateTimeFormatter =
+            DateTimeFormatter.ofPattern(
+                "[yyyy][yy]['-']['/']['.'][' '][MMM][MM][M]['-']['/']['.'][' '][dd][d][[' '][G]][[' ']['T']HH:mm[':'ss[.][SSSSSS][SSSSS][SSSS][SSS][' '][z][zzz][Z][O][x][XXX][XX][X][[' '][G]]]]"
+            )
+        private val TIME_FORMATTER: DateTimeFormatter =
+            DateTimeFormatter.ofPattern(
+                "HH:mm[':'ss[.][SSSSSS][SSSSS][SSSS][SSS][' '][z][zzz][Z][O][x][XXX][XX][X]]"
+            )
+    }
+
+    override fun mapDate(value: DateValue, path: List<String>): AirbyteValue {
+        val epochDay = LocalDate.parse(value.value, DATE_TIME_FORMATTER).toEpochDay()
+        return DateValueIntegral(epochDay = epochDay.toInt())
```

Comment:
Is it caught by the exception handler in the parent class, `AirbyteValueIdentityMapper`?

Reply:
Specifically, this implements a feature where conversion failures result in a field being nulled and a "change" of type "nulled" is collected in the record's metadata. We're trying to enforce that feature in exactly one place (the parent class of the mapper) so that every conversion's exceptions can be handled in the same way.